lib/src/io_file.dart

```diff
@@ -142,6 +146,23 @@ class IOFileAdapter {
       name: name,
     );
   }
+
+  Future<IOFile> fromReadStreamGenerator(
+    Stream<Uint8List> Function([int? s, int? e]) openReadStream,
+    int length, {
+    required String name,
+    required DateTime lastModifiedDate,
+    String? contentType,
+  }) async {
+    return _StreamFile(
+      openReadStream,
+      length,
+      contentType: contentType ?? lookupMimeTypeWithDefaultType(name),
+      path: '',
```

Comment:
Consider putting a comment explaining why the path is empty: "It was generated from a BLOB stream"

Reply:
I'll follow your another suggestion.